node-graph/graph-craft/src/document.rs

```diff
@@ -177,6 +203,11 @@ pub enum NodeInput {
 	/// Input that is provided by the parent network to this document node, instead of from a hardcoded value or another node within the same network.
 	Network(Type),
 
+	/// A Rust source code string. Allows us to insert literal Rust code. Only used for GPU compilation.
+	/// We can use this whenever we spin up Rustc. Sort of like inline assembly, but because our language is Rust, it acts as inline Rust.
+	Inline(InlineRust),
+
+	// TODO: Update, remove, or transplant this comment to a new place (ShortCircuit was renamed to ManualComposition, and may also have some relationship to Network?)
```

Comment:
@TrueDoctor @0HyperCube I'm wondering about this comment and what should become of it. I'm also not sure what the relationship is between the former ShortCitcuit and with Network. It seems that, from this enum at least, ShortCircuit wasn't just renamed to ManualComposition but was instead entirely removed?

Maybe we should move the explanation in this comment to another place where ManualComposition does occur? Or otherwise just delete it?

Reply:
`manual_composition` is now specified on the document node. This reduces invalid state, as it makes it impossible for a document node to have 2 ShortCircuit inputs (aka manual_composition inputs).

The concept explained here is similar to the one on the documentation for `pub manual_composition: Option<Type>`. As such you can merge them however you choose @Keavon.